Runtime/CesiumMetadataValue.cs

```diff
@@ -0,0 +1,1825 @@
+using System;
+using Unity.Mathematics;
+using Reinterop;
+using System.Collections.Generic;
+namespace CesiumForUnity
+{
+    /// <summary>
+    /// Represents a value from a property in glTF metadata. The value belong to one of the
+    /// supported property types in EXT_structural_metadata.
+    /// </summary>
+    [ReinteropNativeImplementation("CesiumForUnityNative::CesiumMetadataValueImpl", "CesiumMetadataValueImpl.h", staticOnly: true)]
+    public partial class CesiumMetadataValue
+    {
+        /// <summary>
+        /// The value as an object. This functions similarly to a std::any in C++.
+        /// </summary>
+        /// <remarks>
+        /// If this is intended to hold an integer vecN or matN value, use the appropriate
+        /// CesiumIntN, CesiumUintN, CesiumIntMatN, or CesiumUintMatN structs. Only
+        /// use Unity.Mathematics for vecNs or matNs with floating point components.
+        /// </remarks>
+        internal System.Object objectValue { get; set; }
+
+        #region Getters
+
+        /// <summary>
+        /// The type of the metadata value as defined in the 
+        /// EXT_structural_metadata extension. Some of these types are not 
+        /// accessible from Unity, but can be converted to a close-fitting type.
+        /// </summary>
+        public CesiumMetadataValueType valueType
+        {
+            get
+            {
+                return CesiumMetadataValueType.GetValueType(this.objectValue);
+            }
+        }
+
+        /// <summary>
+        /// Whether the value is empty, i.e., whether it does not actually represent 
+        /// any data. 
+        /// </summary>
+        /// <remarks>
+        /// A CesiumMetadataValue can be empty, similar to how a C# nullable may be null.
+        /// For example, when the raw value of a property matches the property's specified 
+        /// "no data" value, it will return an empty CesiumMetadataValue.
+        /// </remarks>
+        /// <returns>Whether the value is empty.</returns>
+        public bool isEmpty
+        {
+            get { return this.objectValue == null; }
+        }
+        #endregion
+
+        #region Constructors
+        public CesiumMetadataValue() : this(null)
+        { }
+
+        public CesiumMetadataValue(System.Object value)
+        {
+            // For consistent implementation, convert intN to CesiumIntVecN, uintN to CesiumUintVecN,
+            // intNxN to CesiumIntMatNxN, and uintNxN to CesiumUintMatNxN
```

Comment:
In what way does that help the implementation be more consistent? Most likely I just need to keep reading, but the advantage is really not obvious to me at this point.

Reply:
There's no `Int8`, `Int16`, or `Int64` counterparts to `intN`, so I wrote the `CesiumIntVecN` struct to hold these values as well as their component type. I thought it would be a bit more inconvenient to deal with both `intN` and `CesiumIntVecN` cases, so I just moved the `intN` values into a `CesiumIntVecN` with `Int32` component type.